### PR TITLE
added more documentation

### PR DESCRIPTION
**Pull request by guilhermebodin**

### CODE REVIEW

File: docs/src/manual.md [hunk 1]
```diff
@@ -147,13 +147,21 @@ Note that some of MOI constraints can be bridged, see [Bridges](http://www.julia
...
+## Dualize a model
 
+```@docs
+dualize
+```
 
 ## DualOptimizer
 
-You can solve primal problems using its dual formulation.
+You can solve primal problems using its dual formulation.
```
Comment (blegat) [inline docs/src/manual.md]: Trailing space at the end :)

File: docs/src/manual.md [hunk 2]
```diff
@@ -147,13 +147,21 @@ Note that some of MOI constraints can be bridged, see [Bridges](http://www.julia
...
 
 ## DualOptimizer
 
-You can solve primal problems using its dual formulation.
+You can solve primal problems using its dual formulation. 
 ADD EXAMPLE.
+
+## Adding new sets
+
+As Dualization.jl it build entirely on top of MathOptInterface.jl automatically dualize models with custom sets.
```
Comment (blegat) [inline docs/src/manual.md]: it -> is
build -> built

File: src/dualize.jl [hunk 3]
```diff
@@ -72,8 +72,34 @@ end
...
+The function will return a `DualProblem` struct that has the dualized model
+in the default `DualizableModel{Float64}` macro and a `PrimalDualMap{Float64}` for 
+users to identify the links between primal and dual model.
+
+* A `MathOptInterface.ModelLike` and a `DualProblem{T}` 
+
+The function will fill the provided `DualProblem{T}` with the dualized model. 
+This is should only be useful for more advanced use. 
+
+* A `JuMP Model`
```
Comment (blegat) [inline src/dualize.jl]: If you put it between backsticks, it should be valid code so use JuMP.Model

File: docs/src/index.md [hunk 4]
```diff
@@ -1,8 +1,7 @@
 # Dualization.jl Documentation
 
 Dualization.jl is a package written on top of MathOptInterface that allows users to write the dual of a JuMP model automatically.
-This package has two main features, the function dualize that enables users to get a dualized JuMP model and the DualOptimizer that
-enables users to solve a problem providing the solver it's dual version. 
+This package has two main features, the function dualize that enables users to get a dualized JuMP model and the DualOptimizer that enables users to solve a problem providing the solver it's dual version.
```
Comment (chriscoey) [inline docs/src/index.md]:
```suggestion
This package has two main features: the `dualize` function, which enables users to get a dualized JuMP model, and the `DualOptimizer`, which enables users to solve a problem by providing the solver the dual of the problem. 
```

File: docs/src/manual.md (hunk 1 above)
Comment (chriscoey) [inline docs/src/manual.md]:
```suggestion
You can solve a primal problem by using its dual formulation. 
```

File: docs/src/manual.md [hunk 5]
```diff
@@ -147,13 +147,21 @@ Note that some of MOI constraints can be bridged, see [Bridges](http://www.julia
...
 
-You can solve primal problems using its dual formulation.
+You can solve primal problems using its dual formulation. 
 ADD EXAMPLE.
+
+## Adding new sets
+
+As Dualization.jl it build entirely on top of MathOptInterface.jl automatically dualize models with custom sets. 
+To do this, the user needs to define the set and its dual set in MathOptInterface.jl and provide the functions
+`dual_set` and `set_dot` for the set it wants to dualize.
```
Comment (chriscoey) [inline docs/src/manual.md]: it -> the user

File: docs/src/manual.md (hunk 2 above)
Comment (chriscoey) [reply docs/src/manual.md]: this sentence doesn't make sense even with @blegat's suggestions

File: src/dualize.jl [hunk 6]
```diff
@@ -72,8 +72,34 @@ end
 
 # dualize docs
 """
-dualize(model; kwargs...)
+    dualize(args...; kwargs...)
+
+The function dualize works in three differente ways. The user can provide:
```
Comment (chriscoey) [inline src/dualize.jl]:
```suggestion
The `dualize` function works in three different ways. The user can provide:
```

File: src/dualize.jl [hunk 7]
```diff
@@ -72,8 +72,34 @@ end
...
+
+* A `MathOptInterface.ModelLike`
+
+The function will return a `DualProblem` struct that has the dualized model
+in the default `DualizableModel{Float64}` macro and a `PrimalDualMap{Float64}` for
```
Comment (chriscoey) [inline src/dualize.jl]: "has the x in the default y macro" is a bit confusing

File: src/dualize.jl [hunk 8]
```diff
@@ -72,8 +72,34 @@ end
...
+* A `MathOptInterface.ModelLike`
+
+The function will return a `DualProblem` struct that has the dualized model
+in the default `DualizableModel{Float64}` macro and a `PrimalDualMap{Float64}` for 
+users to identify the links between primal and dual model.
+
+* A `MathOptInterface.ModelLike` and a `DualProblem{T}` 
+
+The function will fill the provided `DualProblem{T}` with the dualized model. 
+This is should only be useful for more advanced use.
```
Comment (chriscoey) [inline src/dualize.jl]: this sentence can be rewritten or could just be removed

File: src/dualize.jl [hunk 9]
```diff
@@ -72,8 +72,34 @@ end
...
+
+The function will return a JuMP model with the dual representation of the problem.
+
+* A `JuMP Model` and an optimizer factory
+
+The function will return a JuMP model with the dual representation of the problem with 
+the `OptimizerFactory` attached. The `OptimizerFactory` is the solver and its key arguments
+that users provide in JuMP models, i.e. `with_optimizer(GLPK.Optimizer)`.
+
+On each and everyone of the methods the user can provide the key argument `dual_names`.
```
Comment (chriscoey) [inline src/dualize.jl]:
```suggestion
On each of these methods, the user can provide the keyword argument `dual_names`.
```

File: src/dualize.jl [hunk 10]
```diff
@@ -72,8 +72,34 @@ end
...
+`dual_names` must be a `DualNames` struct that allows users to set more intuitive names
```
Comment (chriscoey) [inline src/dualize.jl]:
```suggestion
`dual_names` must be a `DualNames` struct. It allows users to set more intuitive names 
```

**Pull request merged**